tools/pkglint/lib/rules.ts

```diff
@@ -818,27 +818,6 @@ export class NodeCompatibility extends ValidationRule {
   }
 }
 
-/**
- * Verifies that the ``@types/`` dependencies are correctly recorded in ``devDependencies`` and not ``dependencies``.
- */
-export class NoAtTypesInDependencies extends ValidationRule {
```

Comment:
Can I ask why we don't need this anymore?

Reply:
Because we're violating the rule that this is trying to enforce in the same PR.

The rule says: "`@types/node` may only occur in `devDependencies`" (reasoning: this is only used by TypeScript so you only need it at compile time, not at runtime).

This is *true*. However, a downstream TypeScript compilation will read the `.d.ts` files generated by this library, and they will contain functions like `function formatDiff(outputStream: NodeJS.Writeable)`. How is a downstream TypeScript compilation to know what `NodeJS.Writeable` means? It needs `@types/node` to know that (and in fact the `.d.ts` file will also contain `import { NodeJS } from '@types/node'` at the top).

So, even though you don't need `@types/node` at *runtime*, you definitely still need it as a consumer IF that consumer happens to be using TypeScript. The alternative is adding `@types/node` to `peerDependencies` and adding it to every CONSUMER's `devDependencies`.

Might also work, involves a lot more changes to our repo (and runs into the NPM6/NPM7 dilemma again for transitive dependencies).